From f04eb08970790184f2d4c51bba2f27ff88ee0a9d Mon Sep 17 00:00:00 2001
From: Lin Xie <lin.xie@intel.com>
Date: Wed, 24 Apr 2013 16:29:35 +0800
Subject: [PATCH 1/6] hwcomposer: Add a new flag HWC_TRICK_MODE.

Add a new flag HWC_TRICK_MODE for layer in hwcomposer.
Signed-off-by: Lin Xie <lin.xie@intel.com>

BZ: 102566
Category: Device Enablement
Domain: AOSP-hardware
Severity: High
Origin: Internal
Upstream-candidate: Yes

Change-Id: I9b2ae36995bb15786d2a39b1ec3486ce4007c1db
Orig-Change-Id: I077555129b97b4376b3ffbd3d2aad483c3a1af3c
Reviewed-on: http://android.intel.com:8080/104307
Reviewed-by: Qiu, Junhai <junhai.qiu@intel.com>
Reviewed-by: Xie, Lin <lin.xie@intel.com>
Reviewed-by: Wang, Lili A <lili.a.wang@intel.com>
Reviewed-by: Ding, Haitao <haitao.ding@intel.com>
Tested-by: Ding, Haitao <haitao.ding@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
---
 include/hardware/hwcomposer_defs.h |    8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/include/hardware/hwcomposer_defs.h b/include/hardware/hwcomposer_defs.h
index 9a52436..b7a51e2 100644
--- a/include/hardware/hwcomposer_defs.h
+++ b/include/hardware/hwcomposer_defs.h
@@ -86,7 +86,13 @@ enum {
      * composition type of this layer, then the hwcomposer will allow async
      * position updates to this layer via setCursorPositionAsync().
      */
-    HWC_IS_CURSOR_LAYER = 0x00000002
+    HWC_IS_CURSOR_LAYER = 0x00000002,
+
+    /*
+     * HWC_TRICK_MODE indicates the layer needs to be rendered on the
+     * display device. Currently it is used on hdmi extend video mode.
+     */
+    HWC_TRICK_MODE = 0x10000000,
 };
 
 /*
-- 
1.7.10.4

